src/client/Microsoft.Identity.Client/Instance/GenericAuthority.cs

```diff
@@ -0,0 +1,59 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Collections.Concurrent;
+using System.Threading;
+using Microsoft.IdentityModel.Protocols;
+using Microsoft.IdentityModel.Protocols.OpenIdConnect;
+
+namespace Microsoft.Identity.Client.Instance
+{
+    internal class GenericAuthority : Authority
+    {
+        internal const string OpenIdConfigurationEndpointSuffix = ".well-known/openid-configuration";
+        private static readonly ConcurrentDictionary<string, OpenIdConnectConfiguration> s_openIdConnectConfigurations = new();
```

Comment:
What is this dictionary used for?

Reply:
@brentschmaltz This dictionary is used to cache discovery responses I'm planning to replace it with either a MemoryCache (but i did not found a mono implementation yet) or a MSAL cache (I did not explored this track yet).

Reply:
Ok!